connector/avro/src/test/scala/org/apache/spark/sql/avro/AvroSuite.scala

```diff
@@ -2252,6 +2252,151 @@ abstract class AvroSuite
     """.stripMargin)
   }
 
+  private def checkSparkSchemaEquals(
+      avroSchema: String, expectedSchema: StructType, recursiveFieldMaxDepth: Int): Unit = {
+    val sparkSchema =
+      SchemaConverters.toSqlType(
+        new Schema.Parser().parse(avroSchema), false, "", recursiveFieldMaxDepth).dataType
+
+    assert(sparkSchema === expectedSchema)
+  }
+
+  test("Translate recursive schema - union") {
```

Comment:
Can we also have a test for more complex recursive behavior, partially used as documentation (what the behavior would look like), and in case in the future something is changed?

Reply:
nvm it is included already, let's resolve this